services/wallet/router/filter_test.go

```diff
@@ -0,0 +1,985 @@
+		})
+	}
+}
+
+func TestIsValidForNetworkComplianceV2(t *testing.T) {
+	tests := []struct {
+		name           string
+		route          []*PathV2
+		fromIncluded   map[uint64]bool
+		fromExcluded   map[uint64]bool
+		expectedResult bool
+	}{
+		{
+			name:           "Route with all included chain IDs",
+			route:          []*PathV2{path1, path2},
+			fromIncluded:   map[uint64]bool{1: true, 2: true},
+			fromExcluded:   map[uint64]bool{},
+			expectedResult: true,
+		},
+		{
+			name:           "Route with fromExcluded only",
+			route:          []*PathV2{path1, path2},
+			fromIncluded:   map[uint64]bool{},
+			fromExcluded:   map[uint64]bool{3: false, 4: false},
+			expectedResult: true,
+		},
+		{
+			name:           "Route without excluded chain IDs",
+			route:          []*PathV2{path1, path2},
+			fromIncluded:   map[uint64]bool{1: false, 2: false},
+			fromExcluded:   map[uint64]bool{3: false, 4: false},
+			expectedResult: true,
+		},
+		{
+			name:           "Route with an excluded chain ID",
+			route:          []*PathV2{path1, path3},
+			fromIncluded:   map[uint64]bool{1: false, 2: false},
+			fromExcluded:   map[uint64]bool{3: false, 4: false},
+			expectedResult: false,
+		},
+		{
+			name:           "Route missing one included chain ID",
+			route:          []*PathV2{path1},
+			fromIncluded:   map[uint64]bool{1: false, 2: false},
+			fromExcluded:   map[uint64]bool{},
+			expectedResult: false,
+		},
+		{
+			name:           "Route with no fromIncluded or fromExcluded",
+			route:          []*PathV2{path1, path2},
+			fromIncluded:   map[uint64]bool{},
+			fromExcluded:   map[uint64]bool{},
+			expectedResult: true,
+		},
+		{
+			name:           "Empty route",
+			route:          []*PathV2{},
+			fromIncluded:   map[uint64]bool{1: false, 2: false},
+			fromExcluded:   map[uint64]bool{3: false, 4: false},
+			expectedResult: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := isValidForNetworkComplianceV2(tt.route, tt.fromIncluded, tt.fromExcluded)
+			assert.Equal(t, tt.expectedResult, result)
+		})
+	}
+}
+
+func TestHasSufficientCapacityV2(t *testing.T) {
+	tests := []struct {
+		name             string
+		route            []*PathV2
+		amountIn         *big.Int
+		fromLockedAmount map[uint64]*hexutil.Big
+		expected         bool
+	}{
+		{
+			name:             "All paths meet required amount",
+			route:            []*PathV2{path1, path2, path3},
+			amountIn:         big.NewInt(600),
+			fromLockedAmount: map[uint64]*hexutil.Big{1: &amount1, 2: &amount2, 3: &amount3},
+			expected:         true,
+		},
+		// TODO: Find out what the expected behaviour for this case should be
+		// I expect false but the test returns true
+		/*
+			{
+				name:             "A path does not meet required amount",
+				route:            []*PathV2{path1, path2, path3},
+				amountIn:         big.NewInt(600),
+				fromLockedAmount: map[uint64]*hexutil.Big{1: &amount1, 2: &amount2, 4: &amount4},
```

Comment:
@Samyoul the reason for this lies in my previous comment, in this case a user entered 100, 300 and 300 on chains 1, 2 and 3 respectively, but locked 100, 200, 400 on chains 1, 2 and 4 respectively, it the function `hasSufficientCapacityV2` you don't count in the value if it's not in the locked amount map, while it should be count.

Reply:
I think that after this change tests that depend on the current `hasSufficientCapacityV2` function should be revised.

Reply:
That's the UI issue. As soon as the user finishes editing an amount for a chain on the "from" side in the Advance tab, the lock should be locked.

Reply:
Ok, so as far as I can see this is a fault with the original logic. I'll make an issue that covers this issue.

Reply:
Thanks!